### ISSUE TITLE
Error "Undefined Index" when fast_404() is enabled in settings.php

### ISSUE THREAD

**Issue by kbahey** [backdrop-contrib/b#214]
When having the fast_404() function call uncommented in settings.php, invoking bee for anything, results in a serious of errors:

```
$ ~/bee/bee.php --site=foo.example.com help
Undefined index: q
/home/someone/sites/test-sites/backdrop/core/includes/bootstrap.inc:3424

Undefined index: q
/home/someone/sites/test-sites/backdrop/core/includes/bootstrap.inc:3426

ini_set(): Headers already sent. You cannot change the session module's ini settings at this time
/home/someone/sites/test-sites/backdrop/sites/foo.example.com/settings.local.php:98

ini_set(): Headers already sent. You cannot change the session module's ini settings at this time
/home/someone/sites/test-sites/backdrop/sites/foo.example.com/settings.local.php:105

ini_set(): Headers already sent. You cannot change the session module's ini settings at this time
/home/someone/sites/test-sites/backdrop/core/includes/bootstrap.inc:979

session_name(): Cannot change session name when headers already sent
/home/someone/sites/test-sites/backdrop/core/includes/bootstrap.inc:991

Bee
Usage: bee [global-options] <command> [options] [arguments]

...
```

Maybe the solution can be to call isset()  on $_GET['q'] at the above lines?

**Comment by yorkshire-pudding** [comment]
Hi @kbahey 

I presume we're talking about the fast_404() line this line here.

```php
/**
 * By default, fast 404s are returned as part of the normal page request
 * process, which will properly serve valid pages that happen to match and will
 * also log actual 404s to the Backdrop log. Alternatively you can choose to
 * return a 404 now by uncommenting the following line. This will reduce server
 * load, but will cause even valid pages that happen to match the pattern to
 * return 404s, rather than the actual page. It will also prevent the Backdrop
 * system log entry. Ensure you understand the effects of this before enabling.
 *
 * To enable this functionality, uncomment the line below.
 */
fast_404();
```
I can reproduce this issue, but given `fast_404()` is disabled by default and users are advised not to unless they understand the consequences, I'm not sure this is something we can address.  Those lines are all in backdrop, not in bee and to change the way bee works to avoid this problem could be quite major.  I have a vague idea why this may be happening due to how bee loads the backdrop bootstrap but I don't think there is an easy fix.

What are your reasons for enabling the `fast_404()` function?

**Comment by kbahey** [comment]
Having fast_404() enabled is crucial for any site that gets substantial traffic. It reduces the CPU and database load considerably by not having Backdrop bootstrap only to return a 404. I have drupal_fast_404() enabled on all my Drupal 7 sites (which I am converting to Backdrop), as well as client sites too. No issues for over a decade, if not more. 

I found a simple change to bee that would solve this issue:

Bee is already fudging many variables because it is run as a CLI, and such things do not exist unless the requests come over HTTP. So adding a fudged variable does the trick. 

Works for me ... 

```
$ git diff
diff --git a/includes/miscellaneous.inc b/includes/miscellaneous.inc
index 9df563a..037fb77 100644
--- a/includes/miscellaneous.inc
+++ b/includes/miscellaneous.inc
@@ -19,6 +19,7 @@ function bee_initialize_server() {
   $_SERVER['SCRIPT_NAME'] = '/index.php';
   $_SERVER['PHP_SELF'] = '/index.php';
   $_SERVER['HTTP_USER_AGENT'] = 'Bee';
+  $_GET['q'] = 'node';
 }
 
 /**
```

**Comment by yorkshire-pudding** [comment]
Hi @kbahey 
Thanks for your explanation and also the suggested fix.  I've tested this locally and it works.  It passes all the tests so I can't see any problems with it.
Are you happy to submit this in a pull request?
If you start the text in the pull request with `Fixes #214` then it will link to this issue and close it once merged.

Thanks
